src/TuDa.CIMS.Web/Components/ShoppingCart/ShoppingCartProductDialog.razor.cs

```diff
@@ -17,11 +19,29 @@ public partial class ShoppingCartProductDialog
     public required AssetItem Product { get; set; }
 
     // Amount of Product
-    public int Amount { get; set; } = 1;
-    private bool IsError => Amount <= 0;
+    public double Amount { get; set; } = 1.0;
+    public int AmountInt { get; set; } = 1;
+
+    /// <summary>
+    /// checks if the amount should be inputted as int or double
+    /// </summary>
+    /// <returns>returns true if it should be an integer</returns>
+    public bool IsInt() => Product is Consumable;
```

Comment:
🤔 Maybe we should also check if `PriceUnit` is per `Piece`
```suggestion
    public bool IsInt() => Product is Consumable || Product.PriceUnit == MeasurementUnits.Piece;
```